src/update-feeds/update-feeds.ts

```diff
@@ -124,8 +138,84 @@ export const runUpdateFeed = async (providerName: string, chain: Chain, chainId:
+export const mergeUrls = (receivedUrls: UrlSet[], freshExistingUrls: UrlSet[]) =>
+  Object.values(groupBy([...receivedUrls, ...freshExistingUrls], 'url')).flatMap(
+    (group) => sortBy(group, 'lastReceivedMs').pop()!
+  );
+
+export const updateDynamicState = (batch: ReadDapiWithIndexResponsesAndChainId) => {
+  batch.map((item) =>
+    updateState((draft) => {
+      const receivedUrls = item.signedApiUrls.flatMap((url) =>
+        item.dataFeed.dataFeeds.flatMap((dataFeed) => ({
+          url: `${url}/${dataFeed.airnodeAddress}`,
+          lastReceivedMs: Date.now(),
+        }))
+      );
+
+      const freshExistingUrls = draft.signedApiUrlStore.filter(
+        (url) => Date.now() - url.lastReceivedMs > 1000 * 60 * SIGNED_URL_EXPIRY_IN_MINUTES
+      );
+
+      // Appends records that don't already exist, updates records that already exist with new timestamps
+      draft.signedApiUrlStore = mergeUrls(receivedUrls.flat(), freshExistingUrls);
+
+      const cachedDapiResponse = draft.dapis[item.dapiName];
+
+      draft.dapis[item.dapiName] = {
+        dataFeed: cachedDapiResponse?.dataFeed ?? item.dataFeed,
+        dataFeedValues: { ...cachedDapiResponse?.dataFeedValues, [item.chainId]: item.dataFeedValue },
+        updateParameters: { ...cachedDapiResponse?.updateParameters, [item.chainId]: item.updateParameters },
+      };
+    })
+  );
+};
+
+export const getFeedsToUpdate = (batch: ReadDapiWithIndexResponsesAndChainId) =>
+  batch
+    .map((dapiResponse) => {
+      const signedData = getStoreDataPoint(dapiResponse.dataFeed.dataFeedId);
+
+      if (signedData === undefined) {
+        return { ...dapiResponse, shouldUpdate: false };
+      }
+
+      const offChainValue = BigNumber.from(signedData.encodedValue);
+      const offChainTimestamp = Number.parseInt(signedData?.timestamp ?? '0', 10);
+      const deviationThreshold = dapiResponse.updateParameters.deviationThresholdInPercentage;
+
+      const shouldUpdate = checkUpdateConditions(
+        dapiResponse.dataFeedValue.value,
+        dapiResponse.dataFeedValue.timestamp,
+        offChainValue,
+        offChainTimestamp,
+        dapiResponse.updateParameters.heartbeatInterval,
+        deviationThreshold
+      );
+
+      if (shouldUpdate) {
+        return {
+          ...dapiResponse,
+          signedData,
+        };
+      }
+
+      return false;
+    })
+    .filter(Boolean);
+
+export const updateFeeds = async (_batch: ReturnType<typeof getFeedsToUpdate>) => {
+  // TODO implement
+  // batch, execute
+};
+
+export const processBatch = async (batch: ReadDapiWithIndexResponsesAndChainId) => {
   logger.debug('Processing batch of active dAPIs', { batch });
-  // TODO: Implement.
+
+  // Start by merging the dynamic state with the state
+  updateDynamicState(batch);
+
+  const feedsToUpdate = getFeedsToUpdate(batch);
+
+  return Promise.allSettled(chunk(feedsToUpdate, FEEDS_TO_UPDATE_CHUNK_SIZE).map(async (feed) => updateFeeds(feed)));
```

Comment:
Remove `FEEDS_TO_UPDATE_CHUNK_SIZE`. We agreed not to chunk these anymore (the whole dAPIs batch will be updated at once).